lib/galaxy/jobs/__init__.py

```diff
@@ -608,27 +608,31 @@ def get_job_tool_configurations(self, ids):
             rval.append(self.default_job_tool_configuration)
         return rval
 
-    def __get_single_item(self, collection):
+    def __get_single_item(self, collection, random_index=None):
```

Comment:
But ``index`` sounds like we are choosing an intentional handler index - it is meant to be sort of a random seed - I sort of think it is just as misleading. I'll change this because I also hate ``random_index`` though. Clearly bumping into on of [the two hard problems](https://twitter.com/codinghorror/status/506010907021828096?lang=en).